### PR TITLE
CB-8480 internal calls broke tags and audit

### PR DESCRIPTION
**Pull request by sodre90**
See detailed description in the commit message.

### CODE REVIEW

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java [hunk 1]
```diff
@@ -285,6 +351,13 @@ UpgradeV4Response checkForClusterUpgradeByName(@PathParam("workspaceId") Long wo
     FlowIdentifier upgradeClusterByName(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name, String imageId,
             @AccountId @QueryParam("accountId") String accountId);
 
+    @POST
+    @Path("internal/{name}/cluster_upgrade")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = STACK_UPGRADE_INTERNAL, nickname = "upgradeClusterByNameInternal")
+    FlowIdentifier upgradeClusterByNameInternal(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name, String imageId,
```
Comment (lnardai) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java:358]: Can't we skip the whole workspace ID from these new API's?

Comment (lnardai) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java:358]: Or use accountId instead?

File: authorization-common/src/main/java/com/sequenceiq/authorization/service/PermissionCheckService.java [hunk 2]
```diff
@@ -54,6 +68,12 @@ public Object hasPermission(ProceedingJoinPoint proceedingJoinPoint) {
         MethodSignature methodSignature = (MethodSignature) proceedingJoinPoint.getSignature();
         LOGGER.debug("Permission check started at {} (method: {})", startTime,
                 methodSignature.getMethod().getDeclaringClass().getSimpleName() + "#" + methodSignature.getMethod().getName());
+        Optional<Object> initiatorUserCrn = reflectionUtil.getParameter(proceedingJoinPoint, methodSignature, InitiatorUserCrn.class);
```
Comment (lnardai) [inline authorization-common/src/main/java/com/sequenceiq/authorization/service/PermissionCheckService.java:71]: Can we check if the endpoint has the @InternalOnly annotation?

**Review by lnardai — APPROVED** [review]

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java (hunk 1 above)
Comment (sodre90) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java:358]: I'd like to, but we have some precheck code for this and fails if there is no workspaceId. We can create a followup task for this but I didn't want to deal with this in this PR.

**Pull request merged**